### ISSUE TITLE
restoring window configurations does not restore scrolling

### ISSUE THREAD

**Issue by sw9** [politza/pdf-tools#225]
I am using eyebrowse-mode to switch between several window configurations. If I have a pdf open in one window configuration, switch to another window configuration, and then switch back, the view of the pdf stays on the correct page but it jumps to the top of the page.

It appears however that vscroll is being saved and restored correctly by window-state-get and window-state-put, because if I manually revert-buffer after restoring the window configuration, the pdf view jumps to the correct spot in the page. I have also tried adding (pdf-view-redisplay t) to the eyebrowse-post-window-switch-hook and this doesn't fix it either.

**Comment by politza** [comment]
This is tricky.  The amount of scrolling is actually stored in
image-mode-winprops-alist, which is part of image-mode.  Both pdf-view
and doc-view use it for displaying PDF pages.  This alist is indexed by
window and contains all the window-specific properties.

So to begin with, image-mode has no concept of window identity beyond
eq, i.e. a window restored by window-state-put looks like a new window
to it.  Luckily it also saves any property modification to the default
properties, which are indexed not by a window, but the value t.

So, in short, restoring a window-configuration should work properly, at
least if the buffer is shown only in one window.  As to why it doesn't,
I have no idea.  I noticed, that if I just evaluate 

M-: t RET

after windows have been restored, it will scroll the PDF where it ought
to be.  So I tend to think that this part is a bug in Emacs.  
amount of scrolling  Can you confirm this ?

**Comment by sw9** [comment]
yes I can confirm that evaluating basically anything with M-: scrolls the pdf to the proper spot. as does making any changes to the window layout (i.e. splitting the window).

**Comment by politza** [comment]
I filled a [bug report](http://debbugs.gnu.org/cgi/bugreport.cgi?bug=24240).  In the meantime,  here is a workaround.

    (defun window-state-put-workaround (&rest _args)
      (run-with-idle-timer 0 nil #'run-window-configuration-change-hook))

    (advice-add 'window-state-put :after #'window-state-put-workaround)
